### ISSUE TITLE
[Bug]: Upgrade Exception while executing repair step `Cache logo dimension to fix size in emails on Outlook`

### ISSUE THREAD

**Issue by lanmarc77** [nextcloud/server#47366]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Using the web updater (as I did many times before) I know get the following error and nextcloud stays in maintenance mode.

```

{"reqId":"aNursuEaVmhMnHgWOXPE","level":1,"time":"2024-08-20T22:27:44+00:00","remoteAddr":"XXXXXXXX","user":"--","app":"updater","method":"GET","url":"/core/ajax/update.php?requesttoken=XXXXXXX","message":"OC\\Repair\\Events\\RepairStepEvent: Repair step: Cache logo dimension to fix size in emails on Outlook","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:129.0) Gecko/20100101 Firefox/129.0","version":"28.0.8.1","data":{"app":"updater"}}

{"reqId":"aNursuEaVmhMnHgWOXPE","level":3,"time":"2024-08-20T22:27:44+00:00","remoteAddr":"XXXXX","user":"--","app":"no app in context","method":"GET","url":"/core/ajax/update.php?requesttoken=XXXXX","message":"Exception while executing repair step Cache logo dimension to fix size in emails on Outlook","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:129.0) Gecko/20100101 Firefox/129.0","version":"28.0.8.1","exception":{"Exception":"OCP\\Files\\NotFoundException","Message":"/appdata_ocn6nxdl32j1/theming/global/images","Code":0,"Trace":[{"file":"/var/www/html/lib/private/Files/Node/Folder.php","line":139,"function":"get","class":"OC\\Files\\Node\\Root","type":"->"},{"file":"/var/www/html/lib/private/Files/SimpleFS/SimpleFolder.php","line":95,"function":"get","class":"OC\\Files\\Node\\Folder","type":"->"},{"file":"/var/www/html/apps/theming/lib/ImageManager.php","line":101,"function":"getFolder","class":"OC\\Files\\SimpleFS\\SimpleFolder","type":"->"},{"file":"/var/www/html/lib/private/Repair/RepairLogoDimension.php","line":47,"function":"getImage","class":"OCA\\Theming\\ImageManager","type":"->"},{"file":"/var/www/html/lib/private/Repair.php","line":129,"function":"run","class":"OC\\Repair\\RepairLogoDimension","type":"->"},{"file":"/var/www/html/lib/private/Updater.php","line":304,"function":"run","class":"OC\\Repair","type":"->"},{"file":"/var/www/html/lib/private/Updater.php","line":144,"function":"doUpgrade","class":"OC\\Updater","type":"->"},{"file":"/var/www/html/core/ajax/update.php","line":187,"function":"upgrade","class":"OC\\Updater","type":"->"}],"File":"/var/www/html/lib/private/Files/Node/Root.php","Line":207,"message":"Exception while executing repair step Cache logo dimension to fix size in emails on Outlook","exception":{},"CustomMessage":"Exception while executing repair step Cache logo dimension to fix size in emails on Outlook"}}

{"reqId":"aNursuEaVmhMnHgWOXPE","level":3,"time":"2024-08-20T22:27:44+00:00","remoteAddr":"XXXXX","user":"--","app":"updater","method":"GET","url":"/core/ajax/update.php?requesttoken=XXXXXXXX","message":"OC\\Repair\\Events\\RepairErrorEvent: Repair error: /appdata_ocn6nxdl32j1/theming/global/images","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:129.0) Gecko/20100101 Firefox/129.0","version":"28.0.8.1","data":{"app":"updater"}}

```

### Steps to reproduce

Using normal webupdate procedure, tried two times.

### Expected behavior

No errors.

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Nginx

### Database engine version

SQlite

### Is this bug present after an update or on a fresh install?

Updated from a MINOR version (ex. 28.0.1 to 28.0.2)

### Are you using the Nextcloud Server Encryption module?

Encryption is Enabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by nickvergessen** [comment]
Interesting, we do a `hasImage` request before and that works...

**Comment by nickvergessen** [comment]
CAn you try the following patch?
```
diff --git a/lib/private/Repair/RepairLogoDimension.php b/lib/private/Repair/RepairLogoDimension.php
index 122da205986..854aeb3ab07 100644
--- a/lib/private/Repair/RepairLogoDimension.php
+++ b/lib/private/Repair/RepairLogoDimension.php
@@ -9,6 +9,8 @@ declare(strict_types=1);
 namespace OC\Repair;
 
 use OCA\Theming\ImageManager;
+use OCP\Files\NotFoundException;
+use OCP\Files\NotPermittedException;
 use OCP\IConfig;
 use OCP\Migration\IOutput;
 use OCP\Migration\IRepairStep;
@@ -44,9 +46,18 @@ class RepairLogoDimension implements IRepairStep {
                        return;
                }
 
-               $simpleFile = $imageManager->getImage('logo', false);
-
-               $image = @imagecreatefromstring($simpleFile->getContent());
+               try {
+                       try {
+                               $simpleFile = $imageManager->getImage('logo', false);
+                               $image = @imagecreatefromstring($simpleFile->getContent());
+                       } catch (NotFoundException|NotPermittedException) {
+                               $simpleFile = $imageManager->getImage('logo');
+                               $image = false;
+                       }
+               } catch (NotFoundException|NotPermittedException) {
+                       $output->info('Theming is not used to provide a logo');
+                       return;
+               }
 
                $dimensions = '';
                if ($image !== false) {
```

**Comment by smartissimo** [comment]
Hey, I am getting this error message:

`root@mail:/var/www/vhosts/xxxxxxxxxx.com/httpdocs/nextcloud# patch -p 1 < /var/www/vhosts/xxxxxxxxxx.com/httpdocs/nextcloud/file.patch
patching file lib/private/Repair/RepairLogoDimension.php
patch unexpectedly ends in middle of line
Hunk #2 FAILED at 46.
1 out of 2 hunks FAILED -- saving rejects to file lib/private/Repair/RepairLogoDimension.php.rej
`

**Comment by smartissimo** [comment]
@nickvergessen Can you help me with this issue?

**Comment by lanmarc77** [comment]
I had the same issue but the patch was so small I applied the changes manually.

**Comment by smartissimo** [comment]
Thanks @lanmarc77, resolved it in that way!

**Comment by nickvergessen** [comment]
In the meantime this should be solved already as 29.0.6 was released last week Tuesday
